### ISSUE TITLE
Formula does not calculate when using named ranges to access the spreadsheet. 

### ISSUE THREAD

**Issue by lafar6502** [EPPlusSoftware/EPPlus#1766]
### EPPlus usage

Commercial use (I have a commercial license)

### Environment

Windows/.Net 4.8

### Epplus version

7.4.0, 7.5.0, 7.0.2 too

### Spreadsheet application

Excel

### Description

Attached code fragment and spreadsheet.
I'm using named ranges to set input Width, Height parameters. Expect the calculation result to be in the Area range.
However, in this case the Area value stays equal to 200, which is the original value from Excel file. Looks like calculation is not done.

Workbook calculation mode is set to Automatic, but even if we call ar.Calculate()  the value does not update.
It does update, however, when we do worksheet.Calculate()  - recalculation of the whole worksheet.
Also, when we do .Calculate() on individual cells - this seems to work. Just ExcelNamedRange.Calculate() doesn't seem to work.

The problem started recently, but i'm not sure what triggered it. Might be related with update of other application components to .net 4.8 (from .Net 4.6.1), but it's unclear.

This is quite a dangerous problem - if the calculation silently fails and gives out incorrect results it is very hard to notice that. And potential damage is very high if the calculation is used in business and makes wrong calculations for extended period of time.
Let me know if you were able to reproduce the issue. 

[x3.xlsx](https://github.com/user-attachments/files/18149564/x3.xlsx)

```
[TestMethod]
public void TestNamedRangeFail()
{
    var path = AppDomain.CurrentDomain.BaseDirectory;
    path = Path.Combine(path, "..\\..\\", "x3.xlsx");
    if (!File.Exists(path)) throw new Exception("File not found:" + path);
    var ep = new ExcelPackage(path);
    
    Assert.AreEqual(ep.Workbook.CalcMode, ExcelCalcMode.Automatic);

    var wr = ep.Workbook.Names["Width"];
    var hr = ep.Workbook.Names["Height"];
    var ar = ep.Workbook.Names["Area"];

    wr.Worksheet.SetValue(wr.Address, 5);
    hr.Worksheet.SetValue(hr.Address, 7);

    ar.Calculate(); //no matter if we do ar.Calculate() or not, the value in the range will not update and remains 200.
    //ar.Worksheet.Calculate(); manual calculate on worksheet will solve the problem
    
    var area = ar.Value;
    var area2 = ar.Worksheet.Cells[ar.Address].Value;

    Assert.AreEqual(area, 35);
    Assert.AreEqual(area, area2);
    
}
```

**Comment by JanKallman** [comment]
Thanks for reporting this. I will have a look at this.

**Comment by JanKallman** [comment]
Yes, this is a bug. EPPlus does not calculate defined names referencing an address when calling the `Calculate` method on the name object. 
This only occurs when calling the Calculate method on the Name object and the name reference a cell address. 
A workaround could be to calculate the workbook or the referencing address:
```C#
if(ar.Worksheet!=null && string.IsNullOrEmpty(ar.Address)==false) ar.Worksheet.Cells[ar.Address].Calculate();
//or
ep.Workbook.Calculate();
```
This issue arise in the rewrite of the formula parser in EPPlus 7.
I will provide a fix shortly.
We apologize for the inconvenience.

**Comment by JanKallman** [comment]
Also, please not the EPPlus will not calculate automatically when having CalcMode set to Automatic. This property only affects how Excel will calculate in runtime.

**Comment by lafar6502** [comment]
Thanks Jan, this was a sneaky one because only sometimes i referenced cells via named range, and sometimes it was just by Excel address. And only in former case i got wrong result, which went unnoticed for quite a while. I'm pretty sure i tested the case many versions of EPPlus ago and it seemed to work, but now i'm not so sure what APIs exactly were used then.
And thanks for this clarification about calculate. Wrong assumption on my side.
So, a question: when i update cells via EPPlus (via Worksheet.SetValue method) - does it recalculate dependent cells automatically, or should i always call .Calculate() on the cells before reading their value?
Or call .Calculate() on the whole worksheet? Which is more efficient?

**Comment by JanKallman** [comment]
EPPlus will only calculate the formulas when calling the different `Calculate` methods (on the workbook, worksheet or range level). However, if you have CalcMode set to automatic, Excel will calculate the workbook when it's opened by the user. 
So, Excel will calculate the formulas on open. If you don't need the calculated values in EPPlus, you don't need to calculate them in EPPlus.
An exception is when you create dynamic array formulas in your workbook or if you want the values to appear in protected mode.
If you need EPPlus to calculate the formulas, I would advice to do so before save when you have fully updated the workbook (unless you need the values before of course.).

**Comment by JanKallman** [comment]
Fixed in EPPlus 7.5.3
